File: Easy/349-intersection-of-two-arrays/try-9.cpp
/*
 * Problem #349: Intersection of Two Arrays
 * Difficulty: Easy
 * Submission: Try 9
 * status: Accepted
 * Language: cpp
 * Date: 18/01/2025, 12:51:59
 * Link: https://leetcode.com/problems/intersection-of-two-arrays/
 */

class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        sort(nums1.begin(),nums1.end());
        sort(nums2.begin(),nums2.end());
        vector<int> vec;
        int i =0 ,j = 0;
        while(i<nums1.size() && j < nums2.size()){
            if(nums1[i] == nums2[j]){
                if(vec.size() == 0 || vec.back()!= nums1[i]){
                    vec.push_back(nums1[i]);
                }
                i++;
            }
            else if(nums1[i] > nums2[j]){
                j++;
            }
            else{
                i++;
            }
        }
        return vec;
    }
};

File: Easy/387-first-unique-character-in-a-string/try-9.py
/*
 * Problem #387: First Unique Character in a String
 * Difficulty: Easy
 * Submission: Try 9
 * status: Accepted
 * Language: python3
 * Date: 06/12/2024, 23:11:44
 * Link: https://leetcode.com/problems/first-unique-character-in-a-string/
 */

class Solution:
    def firstUniqChar(self, s: str) -> int:
        l1=[0]*26
        for i in s:
            l1[ord(i)-ord('a')]+=1
        for i in range(len(s)):
            if(l1[ord(s[i])-ord('a')]==1):
                return i
        return -1
            
        
        
